harness/tests/cli/test_auth.py

```diff
@@ -422,6 +423,30 @@ def test_logout(scenario_set: Logout) -> None:
             cli.main(cmd)
 
 
+@pytest.mark.parametrize("active_user", ["alice", "bob", None])
```

Comment:
Request, here, to move those tests that exercise functionality from just one file (and that don't rely on an actual running master) to `harness/tests/determined/...` per [this doc](https://hpe-aiatscale.atlassian.net/wiki/spaces/ENGINEERIN/pages/1351909582/Python+Testing).

Reply:
I'll move it to `harness/tests/...` and change the doc?  But the extra `/determined/` layer is an unhelpful layer of hierarchy I think.

Reply:
Looking at the other tests in this file, these other tests are actually are testing cli logic (at least, as-written).  These tests also happen to be the only real coverage of common/api/authentication.py.  I'll admit that seems wrong but I don't think moving it is actually any better, unless test_logout() and test_login_scenarios() are re-written to be specific to the authentication module, and separate cli-specific unit tests are written to make sure the cli faithfully passes certain variables into the authentication module.

I think that's a good idea, but I'd prefer to put that off until after my [big auth refactor pr](https://github.com/determined-ai/determined/pull/8347), since that pr already had to refactor these tests a bit.  Rewriting them in this PR would mean I have to rewrite them again in that PR.

Reply:
If you'd like to change the doc from `harness/tests/determined/...` to `harness/tests/...`, I support that change. Just when doing it please also move the files currently in `harness/tests/determined` along with.

Reply:
> I think that's a good idea...

Your logic has the right shape to me even if I don't fully get the details. The scenarios strike me as tests of auth, not of the CLI. But you know this better than I do.

Anyway, if this is work you'd like the team get done later, maybe worth a ticket? Otherwise good chance we forget.